# Limpieza de Datos 

In [355]:
def librerias():
    
    '''Empezamos importando las librerìas necesarias para realizar la limpieza de los datos existentes del Dataset '''

import pandas as pd
import numpy as np

print(librerias.__doc__)

Empezamos importando las librerìas necesarias para realizar la limpieza de los datos existentes del Dataset 


In [356]:
def importarData():
   
    ''' Creamos la funciòn para importar nuestro Dataset '''

df= pd.read_csv('data_act_01.csv')

print(importarData.__doc__)

 Creamos la funciòn para importar nuestro Dataset 


# PRIMER ERROR

El primer error encontrado al ejecutar el dataset como se puede ver es que los datos se encuentran en una sola columna, esto se debe a que los datos se encuentran separados por un punto y coma(;) y no por una coma(,) como debería ser normalmente.

In [357]:
def imprDatos():
   
    ''' Enviamos a imprimir los datos existentes dentro del Dataset '''

print(imprDatos.__doc__)

df

 Enviamos a imprimir los datos existentes dentro del Dataset 


CrimeId;OriginalCrimeTypeName;OffenseDate;CallTime;CallDateTime;Disposition;Address;City;State;AgencyId;Range;AddressType
0      160903280;Assault / Battery;2016-03-30T00:00:0...                                                                       
1      160912272;Homeless Complaint;2016-03-31T00:00:...                                                                       
2      160912590;Susp Info;2016-03-31T00:00:00;16:49;...                                                                       
3      160912801;Report;2016-03-31T00:00:00;17:38;201...                                                                       
4      160912811;594;2016-03-31T00:00:00;17:42;2016-0...                                                                       
...                                                  ...                                                                       
10046  160964210;Well Being Check;2016-04-05T00:00:00...                                                                       
10047  160964216;Traffic Stop;2016-04-05T00:00:00;23:...                                                                       
10048  160964227;Traffic Stop;2016-04-05T00:00:00;23:...                                                                       
10049  160964229;Passing Call;2016-04-05T00:00:00;23:...                                                                       
10050  160964249;Traffic Stop;2016-04-05T00:00:00;23:...                                                                       

[10051 rows x 1 columns]

# Corrección del Error

Para corregir el error presentado en el dataset se utilizó el parámetro sep donde se le especifíca el delimitador con el cual cuenta el archivo. Como se sabe este parámetro nos ayuda a separar los espacios que por defecto Python nos presenta. Y en este caso se le establece al separador que al encontrar un punto y coma en cada uno de los datos los separe, logrando de esta manera presentar los datos correctamente separados.

In [358]:
def impData():
   
    ''' Creamos nuevamente la funciòn para importar nuestro Dataset agregando el parámetro sep, el cual nos permite separar los datos por espacios '''

df= pd.read_csv('data_act_01.csv', sep= ';')

print(impData.__doc__)

 Creamos nuevamente la funciòn para importar nuestro Dataset agregando el parámetro sep, el cual nos permite separar los datos por espacios 


In [359]:
def imprDatos():

    '''Se realiza una función mediante el uso del método head,el cual nos imprime los primeros datos del Dataset '''

print(imprDatos.__doc__)
df.head(10)

Se realiza una función mediante el uso del método head,el cual nos imprime los primeros datos del Dataset 


CrimeId OriginalCrimeTypeName          OffenseDate CallTime  \
0  160903280     Assault / Battery  2016-03-30T00:00:00    18:42   
1  160912272    Homeless Complaint  2016-03-31T00:00:00    15:31   
2  160912590             Susp Info  2016-03-31T00:00:00    16:49   
3  160912801                Report  2016-03-31T00:00:00    17:38   
4  160912811                   594  2016-03-31T00:00:00    17:42   
5  160913003                 Ref'd  2016-03-31T00:00:00    18:29   
6  160913050    Homeless Complaint  2016-03-31T00:00:00    18:43   
7  160913056    Homeless Complaint  2016-03-31T00:00:00    18:47   
8  160913078  Agg Assault / Adw Dv  2016-03-31T00:00:00    18:52   
9  160913103            Encampment  2016-03-31T00:00:00    18:57   

          CallDateTime Disposition                   Address           City  \
0  2016-03-30T18:42:00         REP   100 Block Of Chilton Av  San Francisco   
1  2016-03-31T15:31:00         GOA   2300 Block Of Market St  San Francisco   
2  2016-03-31T16:49:00         GOA   2300 Block Of Market St  San Francisco   
3  2016-03-31T17:38:00         GOA       500 Block Of 7th St  San Francisco   
4  2016-03-31T17:42:00         REP        Beale St/bryant St  San Francisco   
5  2016-03-31T18:29:00         GOA           16th St/pond St  San Francisco   
6  2016-03-31T18:43:00         ADV    Berwick Pl/harrison St  San Francisco   
7  2016-03-31T18:47:00         HAN    Florida St/mariposa St  San Francisco   
8  2016-03-31T18:52:00          ND  100 Block Of Genebern Wy  San Francisco   
9  2016-03-31T18:57:00         ADV   2700 Block Of Folsom St  San Francisco   

  State AgencyId  Range      AddressType  
0    CA        1    NaN  Premise Address  
1    CA        1    NaN  Premise Address  
2    CA        1    NaN  Premise Address  
3    CA        1    NaN  Premise Address  
4    CA        1    NaN     Intersection  
5    CA        1    NaN     Intersection  
6    CA        1    NaN     Intersection  
7    CA        1    NaN     Intersection  
8    CA        1    NaN  Premise Address  
9    CA        1    NaN  Premise Address

# Segundo Error

El segundo error encontrado dentro del dataset son valores nulos dentro de las columnas City, States y Range. Mediante el uso de la función isnull podemos observar el número exacto de valores faltantes dentro de las columnas.

In [360]:
def ValNulos():

    '''Mediante el uso del método isnull visualizamos el total de valores nulos existentes en cada una de las columnas '''

print(ValNulos.__doc__)
df.isnull().sum()


Mediante el uso del método isnull visualizamos el total de valores nulos existentes en cada una de las columnas 


CrimeId                      0
OriginalCrimeTypeName        0
OffenseDate                  0
CallTime                     0
CallDateTime                 0
Disposition                  0
Address                      0
City                       321
State                        3
AgencyId                     0
Range                    10051
AddressType                  0
dtype: int64

# Corrección del Error

Para corregir el error encontrado se hace uso de la función fillna la cual reemplaza los valores faltantes o nulos por valores en cero, en el conjunto de datos existen valores nulos dentro de las columnas City, State y Range tal como se observa. Al implementar esta función podemos ver que donde existían valores NaN los reemplazo por valores en 0, ya que al contar con valores nulos nos puede generar muchos problemas a la hora de realizar un análisis de nuestros datos.

In [361]:
def RempVal():

    '''Creamos una funciòn la cual nos ayudarà a reemplazar los valores vacios que existen en las columnas City, State y Range por valores en Cero, ya que al tener valores nulos nos puede generar muchos problemas a la hora de realizar un análisis de nuestros datos.  '''


df[['City', 'Range','State']] = df[['City', 'Range','State']].fillna(value=0)
    
print(RempVal.__doc__)

Creamos una funciòn la cual nos ayudarà a reemplazar los valores vacios que existen en las columnas City, State y Range por valores en Cero, ya que al tener valores nulos nos puede generar muchos problemas a la hora de realizar un análisis de nuestros datos.  


In [362]:
def impDatos():
   
    '''Enviamos a imprimir nuevamente los datos para verificar que se reemplazaron los valores nulos por valores en 0 '''

print(impDatos.__doc__)

df

Enviamos a imprimir nuevamente los datos para verificar que se reemplazaron los valores nulos por valores en 0 


CrimeId OriginalCrimeTypeName          OffenseDate CallTime  \
0      160903280     Assault / Battery  2016-03-30T00:00:00    18:42   
1      160912272    Homeless Complaint  2016-03-31T00:00:00    15:31   
2      160912590             Susp Info  2016-03-31T00:00:00    16:49   
3      160912801                Report  2016-03-31T00:00:00    17:38   
4      160912811                   594  2016-03-31T00:00:00    17:42   
...          ...                   ...                  ...      ...   
10046  160964210      Well Being Check  2016-04-05T00:00:00    23:38   
10047  160964216          Traffic Stop  2016-04-05T00:00:00    23:43   
10048  160964227          Traffic Stop  2016-04-05T00:00:00    23:48   
10049  160964229          Passing Call  2016-04-05T00:00:00    23:48   
10050  160964249          Traffic Stop  2016-04-05T00:00:00    23:54   

              CallDateTime   Disposition                  Address  \
0      2016-03-30T18:42:00           REP  100 Block Of Chilton Av   
1      2016-03-31T15:31:00           GOA  2300 Block Of Market St   
2      2016-03-31T16:49:00           GOA  2300 Block Of Market St   
3      2016-03-31T17:38:00           GOA      500 Block Of 7th St   
4      2016-03-31T17:42:00           REP       Beale St/bryant St   
...                    ...           ...                      ...   
10046  2016-04-05T23:38:00           GOA    Hollister Av/hawes St   
10047  2016-04-05T23:43:00           ADV   California St/mason St   
10048  2016-04-05T23:48:00           CIT   Rosella Ct/onondaga Av   
10049  2016-04-05T23:48:00  Not recorded          0 Blk Herbst Rd   
10050  2016-04-05T23:54:00           ADV        Grant Av/geary St   

                City State AgencyId  Range      AddressType  
0      San Francisco    CA        1    0.0  Premise Address  
1      San Francisco    CA        1    0.0  Premise Address  
2      San Francisco    CA        1    0.0  Premise Address  
3      San Francisco    CA        1    0.0  Premise Address  
4      San Francisco    CA        1    0.0     Intersection  
...              ...   ...      ...    ...              ...  
10046  San Francisco    CA        1    0.0     Intersection  
10047  San Francisco    CA        1    0.0     Intersection  
10048  San Francisco    CA        1    0.0     Intersection  
10049  San Francisco    CA        1    0.0     Geo-Override  
10050  San Francisco    CA        1    0.0     Intersection  

[10051 rows x 12 columns]

Al verificar una vez más si existen o no datos nulos podemos ver que las columnas City, State y Range ya no presentar datos nulos.

In [363]:
def ValNul():

    '''Mediante el uso del método isnull visualizamos nuevamente el total de valores nulos, con el fin de verificar que ya no exista ningun valor nulo dentro de las columnas modificadas '''

print(ValNul.__doc__)

df.isnull().sum()

Mediante el uso del método isnull visualizamos nuevamente el total de valores nulos, con el fin de verificar que ya no exista ningun valor nulo dentro de las columnas modificadas 


CrimeId                  0
OriginalCrimeTypeName    0
OffenseDate              0
CallTime                 0
CallDateTime             0
Disposition              0
Address                  0
City                     0
State                    0
AgencyId                 0
Range                    0
AddressType              0
dtype: int64

# Tercer Error

El tercer error presentado es la existencia de datos duplicados dentro de la columna CrimeId. Mediante el uso de la función duplicated podemos verificar el número exacto de datos duplicados dentro de la columna CrimerId.

In [365]:
def DuplDat():

    '''Mediante el uso del método duplicated verficamos el número de datos duplicados dentro de la columna CrimeId   '''

print(DuplDat.__doc__)

df['CrimeId'].duplicated().sum()

Mediante el uso del método duplicated verficamos el número de datos duplicados dentro de la columna CrimeId   


4

In [366]:
def DuplDat():

    '''Creamos un filtro de los datos duplicados de la columna CrimeID y los transformamos en una lista.   '''

print(DuplDat.__doc__)

duplicates = df[df['CrimeId'].duplicated(keep=False)]['CrimeId'].tolist()

Creamos un filtro de los datos duplicados de la columna CrimeID y los transformamos en una lista.   


In [367]:
def ImpDupl():

    '''Visualizamos los datos de los ID duplicados que existen dentro de la columna CrimeId.   '''

print(ImpDupl.__doc__)

duplicated.loc[:, ['CrimeId']]

Visualizamos los datos de los ID duplicados que existen dentro de la columna CrimeId.   


CrimeId
26    160913455
1707  160913455
3792  160913455
7045  160950496
7046  160950496
7047  160950496

# Corrección del Error

Para corregir el error encontrado se crea un set de la lista de los valores, donde se le especifíca mediante el for que se debe recorrer el set de duplicados y debe comparar con cada uno de los datos en el Dataframe. Al encontrar un dato igual este debe sumarle 1 al número de CrimeId, de esta forma evitamos que se den valores duplicados dentro del CrimeId que es donde nos presenta el error.

In [368]:
def FunDup():

    '''Realizamos una función la cual crea un set de la lista de los datos duplicados existentes en la columna CrimeId   '''

set_duplicates = set(duplicates)

print(FunDup.__doc__)



Realizamos una función la cual crea un set de la lista de los datos duplicados existentes en la columna CrimeId   


In [369]:
def FunSet():

    '''Realizamos una función mediante el uso de un for el cual debe recorrer el set de duplicados y debe comparar con cada uno de los datos en el Dataframe. Al encontrar un dato igual este debe sumarle 1 al número de CrimeId, de esta forma evitamos que se den valores duplicados dentro del CrimeId que es donde nos presenta el error. '''

for r in set_duplicates:
    dup = df[df['CrimeId'] == r]
    df.loc[dup.iloc[1,:].name, 'CrimeId'] = df.loc[dup.iloc[1,:].name]['CrimeId'] + 1
                                                                       
print(FunSet.__doc__)


Realizamos una función mediante el uso de un for el cual debe recorrer el set de duplicados y debe comparar con cada uno de los datos en el Dataframe. Al encontrar un dato igual este debe sumarle 1 al número de CrimeId, de esta forma evitamos que se den valores duplicados dentro del CrimeId que es donde nos presenta el error. 


Verificamos una vez más el número de duplicados dentro de la columna mediante la función duplicated y como podemos observar ya no presenta ningún dato duplicado.

In [370]:
def ImprDupl():

    '''Visualizamos nuevamente los datos para verificar que no haya ningun registro duplicado dentro de la columna CrimeId '''

print(ImprDupl.__doc__)

df.duplicated().sum()

Visualizamos nuevamente los datos para verificar que no haya ningun registro duplicado dentro de la columna CrimeId 


0

# Cuarto Error

El cuarto error presentado como se puede visualizar dento de las columnas OffenseDate y CallDateTime contienen la misma información y por ende el mismo formato de fecha y hora.

In [371]:
def FilColmn():

    '''Mediante el uso del método loc seleccionamos las filas y las columnas de las columnas OffenseDate y CallDateTime para poder visualizar de mejor manera el tipo de formato que manejan en la Fecha del delito y la Fecha y hora de la llamada '''

print(FilColmn.__doc__)

df.loc[:, ['OffenseDate', 'CallDateTime']]

Mediante el uso del método loc seleccionamos las filas y las columnas de las columnas OffenseDate y CallDateTime para poder visualizar de mejor manera el tipo de formato que manejan en la Fecha del delito y la Fecha y hora de la llamada 


OffenseDate         CallDateTime
0      2016-03-30T00:00:00  2016-03-30T18:42:00
1      2016-03-31T00:00:00  2016-03-31T15:31:00
2      2016-03-31T00:00:00  2016-03-31T16:49:00
3      2016-03-31T00:00:00  2016-03-31T17:38:00
4      2016-03-31T00:00:00  2016-03-31T17:42:00
...                    ...                  ...
10046  2016-04-05T00:00:00  2016-04-05T23:38:00
10047  2016-04-05T00:00:00  2016-04-05T23:43:00
10048  2016-04-05T00:00:00  2016-04-05T23:48:00
10049  2016-04-05T00:00:00  2016-04-05T23:48:00
10050  2016-04-05T00:00:00  2016-04-05T23:54:00

[10051 rows x 2 columns]

# Corrección del Error

In [372]:
def FormFecDel():

    '''Creamos una función mediante el uso del método to_datetime para modificar el formato de hora y fecha correctamente dentro de la columna OffenseDate(Fecha del delito) '''

print(FormFecDel.__doc__)

df['OffenseDate'] = pd.to_datetime(df['OffenseDate'])

Creamos una función mediante el uso del método to_datetime para modificar el formato de hora y fecha correctamente dentro de la columna OffenseDate(Fecha del delito) 


In [373]:
def FormFecHor():

    '''Creamos una función mediante el uso del método to_datetime para modificar el formato de hora y fecha correctamente dentro de la columna CallDateTime(Fecha y hora de la llamada ) '''

print(FormFecHor.__doc__)

df['CallDateTime'] = pd.to_datetime(df['CallDateTime'])

Creamos una función mediante el uso del método to_datetime para modificar el formato de hora y fecha correctamente dentro de la columna CallDateTime(Fecha y hora de la llamada ) 


In [374]:

def FormHora():

    '''Creamos una función mediante el uso del método to_datetime para modificar el formato de hora correctamente dentro de la columna CallTime, la cual nos presenta el Tiempo de duración de llamada '''

print(FormHora.__doc__)

df['CallTime']= pd.to_datetime(df['CallTime'], format='%H:%M:%S', errors='ignore')


Creamos una función mediante el uso del método to_datetime para modificar el formato de hora correctamente dentro de la columna CallTime, la cual nos presenta el Tiempo de duración de llamada 


In [375]:
df

CrimeId OriginalCrimeTypeName OffenseDate CallTime  \
0      160903280     Assault / Battery  2016-03-30    18:42   
1      160912272    Homeless Complaint  2016-03-31    15:31   
2      160912590             Susp Info  2016-03-31    16:49   
3      160912801                Report  2016-03-31    17:38   
4      160912811                   594  2016-03-31    17:42   
...          ...                   ...         ...      ...   
10046  160964210      Well Being Check  2016-04-05    23:38   
10047  160964216          Traffic Stop  2016-04-05    23:43   
10048  160964227          Traffic Stop  2016-04-05    23:48   
10049  160964229          Passing Call  2016-04-05    23:48   
10050  160964249          Traffic Stop  2016-04-05    23:54   

             CallDateTime   Disposition                  Address  \
0     2016-03-30 18:42:00           REP  100 Block Of Chilton Av   
1     2016-03-31 15:31:00           GOA  2300 Block Of Market St   
2     2016-03-31 16:49:00           GOA  2300 Block Of Market St   
3     2016-03-31 17:38:00           GOA      500 Block Of 7th St   
4     2016-03-31 17:42:00           REP       Beale St/bryant St   
...                   ...           ...                      ...   
10046 2016-04-05 23:38:00           GOA    Hollister Av/hawes St   
10047 2016-04-05 23:43:00           ADV   California St/mason St   
10048 2016-04-05 23:48:00           CIT   Rosella Ct/onondaga Av   
10049 2016-04-05 23:48:00  Not recorded          0 Blk Herbst Rd   
10050 2016-04-05 23:54:00           ADV        Grant Av/geary St   

                City State AgencyId  Range      AddressType  
0      San Francisco    CA        1    0.0  Premise Address  
1      San Francisco    CA        1    0.0  Premise Address  
2      San Francisco    CA        1    0.0  Premise Address  
3      San Francisco    CA        1    0.0  Premise Address  
4      San Francisco    CA        1    0.0     Intersection  
...              ...   ...      ...    ...              ...  
10046  San Francisco    CA        1    0.0     Intersection  
10047  San Francisco    CA        1    0.0     Intersection  
10048  San Francisco    CA        1    0.0     Intersection  
10049  San Francisco    CA        1    0.0     Geo-Override  
10050  San Francisco    CA        1    0.0     Intersection  

[10051 rows x 12 columns]

In [376]:
def FormHora():

    '''Mediante el uso de la función dtypes podemos visulizar que el tipo de dato de las columnas OffenseDate y CallDateTime cambiaron al formato datetime '''
    
print(FormHora.__doc__)

df.dtypes

Mediante el uso de la función dtypes podemos visulizar que el tipo de dato de las columnas OffenseDate y CallDateTime cambiaron al formato datetime 


CrimeId                           int64
OriginalCrimeTypeName            object
OffenseDate              datetime64[ns]
CallTime                         object
CallDateTime             datetime64[ns]
Disposition                      object
Address                          object
City                             object
State                            object
AgencyId                         object
Range                           float64
AddressType                      object
dtype: object

# Quinto Error

El quinto error que se presentó fue que dentro de la columna Range no contaba con ningún tipo de dato, a lo que se entiende que esta columna no representa nada dentro del Dataset. Cabe recalcar que al realizar un shape visualizamos que existen 10051 datos y es el mismo número de valores nulos que presentaba la columna mencionada antes de realizar el reemplazo de datos nulos. Es por esto que se decidió eliminar por completo la columna.


In [364]:
def Shape():

    '''Mediante el uso del método shape visualizamos el valor total de los datos y las columnas con las que cuenta el Dataset  '''

print(Shape.__doc__)

df.shape

Mediante el uso del método shape visualizamos el valor total de los datos y las columnas con las que cuenta el Dataset  


(10051, 12)

In [382]:
def imprDatos():

    '''Enviamos a imprimir los datos del Dataset para verificar los datos nulos que existen dentro de la columna Range'''

print(imprDatos.__doc__)

df

Enviamos a imprimir los datos del Dataset para verificar los datos nulos que existen dentro de la columna Range


CrimeId OriginalCrimeTypeName OffenseDate CallTime  \
0      160903280     Assault / Battery  2016-03-30    18:42   
1      160912272    Homeless Complaint  2016-03-31    15:31   
2      160912590             Susp Info  2016-03-31    16:49   
3      160912801                Report  2016-03-31    17:38   
4      160912811                   594  2016-03-31    17:42   
...          ...                   ...         ...      ...   
10046  160964210      Well Being Check  2016-04-05    23:38   
10047  160964216          Traffic Stop  2016-04-05    23:43   
10048  160964227          Traffic Stop  2016-04-05    23:48   
10049  160964229          Passing Call  2016-04-05    23:48   
10050  160964249          Traffic Stop  2016-04-05    23:54   

             CallDateTime   Disposition                  Address  \
0     2016-03-30 18:42:00           REP  100 Block Of Chilton Av   
1     2016-03-31 15:31:00           GOA  2300 Block Of Market St   
2     2016-03-31 16:49:00           GOA  2300 Block Of Market St   
3     2016-03-31 17:38:00           GOA      500 Block Of 7th St   
4     2016-03-31 17:42:00           REP       Beale St/bryant St   
...                   ...           ...                      ...   
10046 2016-04-05 23:38:00           GOA    Hollister Av/hawes St   
10047 2016-04-05 23:43:00           ADV   California St/mason St   
10048 2016-04-05 23:48:00           CIT   Rosella Ct/onondaga Av   
10049 2016-04-05 23:48:00  Not recorded          0 Blk Herbst Rd   
10050 2016-04-05 23:54:00           ADV        Grant Av/geary St   

                City State AgencyId      AddressType  
0      San Francisco    CA        1  Premise Address  
1      San Francisco    CA        1  Premise Address  
2      San Francisco    CA        1  Premise Address  
3      San Francisco    CA        1  Premise Address  
4      San Francisco    CA        1     Intersection  
...              ...   ...      ...              ...  
10046  San Francisco    CA        1     Intersection  
10047  San Francisco    CA        1     Intersection  
10048  San Francisco    CA        1     Intersection  
10049  San Francisco    CA        1     Geo-Override  
10050  San Francisco    CA        1     Intersection  

[10051 rows x 11 columns]

# Corrección del Error

In [378]:
def elimRange():

    '''Mediante el uso del método drop eliminamos la columna Range, ya que al no tener ningún tipo de dato no es necesaria la limpieza de los datos   '''

print(elimRange.__doc__)

df.drop(['Range'], axis = 'columns', inplace=True)

Mediante el uso del método drop eliminamos la columna Range, ya que al no tener ningún tipo de dato no es necesaria la limpieza de los datos   


In [379]:
def ImpElim():

    '''Visualizamos nuevamente los datos para verificar que se eliminó la columna Range correctamente '''

print(ImpElim.__doc__)

df

Visualizamos nuevamente los datos para verificar que se eliminó la columna Range correctamente 


CrimeId OriginalCrimeTypeName OffenseDate CallTime  \
0      160903280     Assault / Battery  2016-03-30    18:42   
1      160912272    Homeless Complaint  2016-03-31    15:31   
2      160912590             Susp Info  2016-03-31    16:49   
3      160912801                Report  2016-03-31    17:38   
4      160912811                   594  2016-03-31    17:42   
...          ...                   ...         ...      ...   
10046  160964210      Well Being Check  2016-04-05    23:38   
10047  160964216          Traffic Stop  2016-04-05    23:43   
10048  160964227          Traffic Stop  2016-04-05    23:48   
10049  160964229          Passing Call  2016-04-05    23:48   
10050  160964249          Traffic Stop  2016-04-05    23:54   

             CallDateTime   Disposition                  Address  \
0     2016-03-30 18:42:00           REP  100 Block Of Chilton Av   
1     2016-03-31 15:31:00           GOA  2300 Block Of Market St   
2     2016-03-31 16:49:00           GOA  2300 Block Of Market St   
3     2016-03-31 17:38:00           GOA      500 Block Of 7th St   
4     2016-03-31 17:42:00           REP       Beale St/bryant St   
...                   ...           ...                      ...   
10046 2016-04-05 23:38:00           GOA    Hollister Av/hawes St   
10047 2016-04-05 23:43:00           ADV   California St/mason St   
10048 2016-04-05 23:48:00           CIT   Rosella Ct/onondaga Av   
10049 2016-04-05 23:48:00  Not recorded          0 Blk Herbst Rd   
10050 2016-04-05 23:54:00           ADV        Grant Av/geary St   

                City State AgencyId      AddressType  
0      San Francisco    CA        1  Premise Address  
1      San Francisco    CA        1  Premise Address  
2      San Francisco    CA        1  Premise Address  
3      San Francisco    CA        1  Premise Address  
4      San Francisco    CA        1     Intersection  
...              ...   ...      ...              ...  
10046  San Francisco    CA        1     Intersection  
10047  San Francisco    CA        1     Intersection  
10048  San Francisco    CA        1     Intersection  
10049  San Francisco    CA        1     Geo-Override  
10050  San Francisco    CA        1     Intersection  

[10051 rows x 11 columns]

# Archivo Json

Una vez realizada la limpieza de los datos del Dataset seleccionado se procede a crear un documento JSON, el cual contiene toda la información de las columnas y filas modificadas.

In [383]:
def DocJSON():

    '''Creamos una función mediante el uso del método to_json, el cual nos permite crear  convertir un DataFrame de pandas en una lista de diccionarios de los datos limpios del Dataset'''

print(DocJSON.__doc__)

df.to_json('data_act_01_pFormateada.json', orient='records', lines=True)

Creamos una función mediante el uso del método to_json, el cual nos permite crear  convertir un DataFrame de pandas en una lista de diccionarios de los datos limpios del Dataset
